src/Raven.Client/Documents/Queries/Facets/FacetBase.cs

```diff
@@ -17,6 +17,8 @@ protected FacetBase()
 
         public Dictionary<FacetAggregation, string> Aggregations { get; set; }
 
+        public Parameters FacetParameters { get; set; }
```

Comment:
Wouldn't it be better to delay the conversion of Facet (either Facet, Facet<T>, RangeFacet or RangeFacet<T>) to the last possible point (where FacetToken is created). You would not have to drag the parameters along the way, then rewrite them to query parameters, rewrite the expressions, etc. This would cause much less trouble.